Toolkit/NewAppDeployToolkit/Functions/Main/Close-InstallationProgress.ps1

```diff
@@ -16,87 +15,120 @@ Function Close-InstallationProgress {
-			}
-			$Timeout -= 1
-			Start-Sleep -Seconds 1
+
+	process {
+		if ($deployModeSilent) {
+			Write-Log -Message "Bypassing Close-InstallationProgress [Mode: $deployMode]" -Source $CmdletName
+			return
 		}
-		# Return if we still have no window
+
+		# Check whether the window has been created
 		if (-not $script:ProgressSyncHash.Window.IsInitialized) {
-			Write-Log -Message "The installation progress dialog was not created within $WaitingTime seconds." -Source ${CmdletName} -Severity 2
-			return
+			Write-Log -Message "The installation progress dialog does not exist. Waiting up to $WaitingTime seconds..." -Source ${CmdletName}
+
+			#wait for up to $WaitingTime seconds if the window does not exist
+	
+			$Timeout = $WaitingTime
+			while ($Timeout -gt 0) {
+				if ($script:ProgressSyncHash.Window.IsInitialized) {
+					break
+				}
+				$Timeout -= 1
+				Start-Sleep -Seconds 1
+			}
+			
+			# Return if we still have no window
+			if (-not $script:ProgressSyncHash.Window.IsInitialized) {
+				Write-Log -Message "The installation progress dialog was not created within $WaitingTime seconds." -Source ${CmdletName} -Severity 2
+				return
+			}
 		}
+		
 		# If the thread is suspended, resume it
 		if ($script:ProgressSyncHash.Window.Dispatcher.Thread.ThreadState -band [system.threading.threadstate]::Suspended) {
-			Write-Log -Message 'The thread for the installation progress dialog is suspended. Resuming the thread.' -Source ${CmdletName}
+			Write-Log -Message 'The thread for the installation progress dialog is suspended. Resuming the thread.' -Source $CmdletName
 			try {
 				$script:ProgressSyncHash.Window.Dispatcher.Thread.Resume()
-			}
-			catch {
-				Write-Log -Message 'Failed to resume the thread for the installation progress dialog.' -Source ${CmdletName} -Severity 2
+			} catch {
+				Write-Log -Message 'Failed to resume the thread for the installation progress dialog.' -Source $CmdletName -Severity 2
 			}
 		}
+
 		# If the thread is changing its state, wait
-		[int]$Timeout = 0
-		while ((($script:ProgressSyncHash.Window.Dispatcher.Thread.ThreadState -band [system.threading.threadstate]::Aborted) -or ($script:ProgressSyncHash.Window.Dispatcher.Thread.ThreadState -band [system.threading.threadstate]::AbortRequested) -or ($script:ProgressSyncHash.Window.Dispatcher.Thread.ThreadState -band [system.threading.threadstate]::StopRequested) -or ($script:ProgressSyncHash.Window.Dispatcher.Thread.ThreadState -band [system.threading.threadstate]::Unstarted) -or ($script:ProgressSyncHash.Window.Dispatcher.Thread.ThreadState -band [system.threading.threadstate]::WaitSleepJoin)) -and ($Timeout -le $WaitingTime)) {
-			if (-not $Timeout) {
-				Write-Log -Message "The thread for the installation progress dialog is changing its state. Waiting up to $WaitingTime seconds..." -Source ${CmdletName} -Severity 2
+		if (($script:ProgressSyncHash.Window.Dispatcher.Thread.ThreadState -band [system.threading.threadstate]::Aborted)
```

Comment:
Need to remember that this needs to be looked at in future

Reply:
im gonna add a #TODO

Reply:
made an issue #6